src/main/java/com/google/sps/data/Event.java

```diff
@@ -0,0 +1,184 @@
+    this.date = date;
+    this.startTime = startTime;
+    this.travelTime = travelTime;
+    this.endTime = calculateEndTime(timeAtLocation);
+
+    createCalendarTimes();
+  }
+ 
+  /**
+   * Constructor that assume one hour default spent at location
+   * 
+   * @param name date from user input travel day. Cannot be null.
+   * @param address time to to start or end event. Must be within bounds
+   * @param date day of travel
+   * @param startTime start of activing
+   * @param travelTime time spent traveling to next location
+   */
+  public Event(String name, String address, String date, int startTime, int travelTime) {
+    this.name = name;
+    this.address = address;
+    this.date = date;
+    this.startTime = startTime;
+    this.travelTime = travelTime;
+    this.endTime = calculateEndTime(HOUR);
+
+    createCalendarTimes();
+  }
+
+  // function that sets start and end Time with correct string format
+  private void createCalendarTimes() {
+    this.strStartTime = createStrTime(this.date, this.startTime);
+    this.strEndTime = createStrTime(this.date, this.endTime);
+  }
+
+  /**
+   * function that builds string w correct format for calendar-script.js
+   * 
+   * @param date date from user input travel day. Cannot be null.
+   * @param time time to to start or end event. Must be within bounds
+   */
+  private static String createStrTime(String date, int time) {
+    if (date == null) {
+      throw new NullPointerException("Date cannot be null");
+    }
+
+    if (time < MIN_POSSIBLE_TIME) {
+      throw new IllegalArgumentException("Time cannot be less than 0");
+    }
+
+    if (time >= MINUTES_IN_A_DAY) {
+      throw new IllegalArgumentException("Time cannot be more than a 24hrs");
+    }
+
+    String output = "";
+    output += date + "T" + Integer.toString(time).substring(0, 2) + ":" + 
+      Integer.toString(time).substring(2, 4) + ":00";
+    return output;
+  }
+
+  /**
+   * function that calculates endTime given start and timeSpent
+   * 
+   * @param timeAtLocation time that the user spends at a POI
+   */
+  private int calculateEndTime(int timeAtLocation) {
+    return this.startTime + Integer.parseInt(convertToFormat(timeAtLocation));
+  }
+
+  /**
+   * function that converts mintues into hhmm format
+   * 
+   * @param time time in minutes
+   */
+  public static String convertToFormat(int time) {
+    if (time < MIN_POSSIBLE_TIME) {
+      throw new IllegalArgumentException("Time cannot be less than 0");
+    }
+
+    if (time >= MINUTES_IN_A_DAY) {
+      throw new IllegalArgumentException("Time cannot be more than a 24hrs");
+    }
+
+    String numHours = Integer.toString(time / 60);
+    String numMins = Integer.toString(time % 60);
+
+    // add 0's to front of number if neccessary
+    if (Integer.parseInt(numHours) < 10) {
+      numHours = "0" + numHours;
+    } 
+
+    if (Integer.parseInt(numMins) < 10) {
+      numMins = "0" + numMins;
+    }
+
+    return numHours + numMins;
+  }
```

Comment:
Why is this function `static`? In what situations will we use this outside the class? (Not saying it shouldn't be `static`, just genuinely curious.)

Reply:
A `static` function cannot access non-static fields of the enclosing class. But, otoh, a client can call static functions without an instance of the class. Eg: your code could call `Event.convertToFormat(1234)` without having to construct an Event object. So, as in this case, `static` is often used for stateless utility functions ("stateless" is another way of saying it doesn't modify class fields).

@adamfweidman can probably give more context on how/why he decided to have this be its own method and why he wanted it to be public.